crates/net/network/src/manager.rs

```diff
@@ -667,6 +668,23 @@ pub enum NetworkEvent {
         /// The status of the peer to which a session was established.
         status: Status,
     },
+    /// Event emitted when peers are added/dropped or
+    /// when a message is sent or received on a peer connection
+    /// See: https://github.com/ethereum/go-ethereum/blob/53d1ae096ac0515173e17f0f81a553e5f39027f7/p2p/peer.go#L94
+    PeerEvent {
+        /// The type of event
+        event_type: PeerEventType,
+        /// The id of the peer which triggered this event
+        peer: PeerId,
+        /// If peer was dropped, the disconnect reason
+        error: Option<DisconnectReason>,
+        /// Protocol of peer
+        protocol: Option<PeerProtocolsInfo>,
+        /// Message sent/received by the peer, if any
+        message: Option<ProtocolMessage>,
+        /// Network information
+        network_info: Option<PeerNetworkInfo>,
+    },
```

Comment:
I'd like to start with dedicated event variants for `PeerAdded(PeerId)` and `PeerRemoved(PeerId)`

These can be included in the `NetworkEvent`, for message monitoring, I'm not sure yet, because there'll be a lot of them and since we basically only need them for the admin RPC subscription, we can likely handle them separately.

Reply:
Do the `PeerAdded` and `PeerRemoved` variants only need `peerId` (without any other members like `error`, `protocol`, etc. ? If so, the `PeerEventType` enum I added also doesn't make much sense. Should I remove that then?

Reply:
> the PeerEventType enum I added also doesn't make much sense. Should I remove that then?

yes, we only want `PeerAdded` and `PeerRemoved` instead here.

Otherwise, this event type becomes very noisy if it includes the messages etc. we can handle these separately elsewhere.